src/matrix/kernelkroneckermat.jl

```diff
@@ -4,56 +4,54 @@
 using .Kronecker: Kronecker
 
 export kernelkronmat
-export kronecker_kernelmatrix
 
 @doc raw"""
-    kernelkronmat(κ::Kernel, X::AbstractVector{<:Real}, dims::Int) -> KroneckerPower
+    kernelkronmat(k::Kernel, x::AbstractVector{<:Real}, d::Int)
 
-Return a `KroneckerPower` matrix on the `D`-dimensional input grid constructed by ``\otimes_{i=1}^D X``,
-where `D` is given by `dims`.
+Compute the [`kernelmatrix`](@ref) for kernel `k` on the `d`-dimensional grid ``\otimes_{i=1}^d x``
+as a lazy Kronecker product.
 
 !!! warning
-
-    Require `Kronecker.jl` and for `iskroncompatible(κ)` to return `true`.
+    You have to load [Kronecker.jl](https://github.com/MichielStock/Kronecker.jl) to use this function.
+    Additionally, `iskroncompatible(k)` has to be `true`.
 """
-function kernelkronmat(κ::Kernel, X::AbstractVector{<:Real}, dims::Int)
-    checkkroncompatible(κ)
-    K = kernelmatrix(κ, X)
-    return Kronecker.kronecker(K, dims)
+function kernelkronmat(k::Kernel, x::AbstractVector{<:Real}, d::Int)
+    checkkroncompatible(k)
+    K = kernelmatrix(k, x)
+    return Kronecker.kronecker(K, d)
 end
 
 @doc raw"""
+    kernelkronmat(k::Kernel, x::AbstractVector{<:AbstractVector})
```

Comment:
```suggestion
    kernelkronmat(k::Kernel, x::AbstractVector{<:AbstractVector}) -> KroneckerProduct
```
I think it's very helpful to know what the return type is